test/e2e/apps/statefulset.go

```diff
@@ -577,11 +578,42 @@ var _ = SIGDescribe("StatefulSet", func() {
 		framework.ConformanceIt("Scaling should happen in predictable order and halt if any stateful pod is unhealthy [Slow]", func() {
 			psLabels := klabels.Set(labels)
 			w := &cache.ListWatch{
+				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+					options.LabelSelector = psLabels.AsSelector().String()
+					// Set ResourceVersion to "" to ensure we start at now and won't
+					// miss needed or unintetionally include some past events.
+					options.ResourceVersion = ""
+					return f.ClientSet.CoreV1().Pods(ns).List(context.TODO(), options)
+				},
 				WatchFunc: func(options metav1.ListOptions) (i watch.Interface, e error) {
 					options.LabelSelector = psLabels.AsSelector().String()
 					return f.ClientSet.CoreV1().Pods(ns).Watch(context.TODO(), options)
 				},
 			}
+
+			// Verify that statuful set will be scaled up in order.
+			wg := sync.WaitGroup{}
+			var orderErr error
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				expectedOrder := []string{ssName + "-0", ssName + "-1", ssName + "-2"}
+				ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), statefulSetTimeout)
+				defer cancel()
+
+				_, orderErr = watchtools.UntilWithSync(ctx, lw, &v1.Pod{}, nil, func(event watch.Event) (bool, error) {
```

Comment:
move this after `Initializing watcher for selector` and use watchtools.Until, starting at pl.ResourceVersion

Reply:
done